src/Entity/Repository/OfferingRepository.php

```diff
@@ -89,6 +90,43 @@ public function findDTOsBy(array $criteria, array $orderBy = null, $limit = null
         return array_values($offeringDTOs);
     }
 
+    public function getOfferingsForTeachingReminders(int $daysInAdvance, array $schoolIds): array
+    {
+        $now = time();
+        $startDate = new \DateTime();
+        $startDate->setTimezone(new \DateTimeZone('UTC'));
+        $startDate->setTimestamp($now);
+        $startDate->modify("midnight +{$daysInAdvance} days");
+
+        $daysInAdvance++;
+        $endDate = new \DateTime();
+        $endDate->setTimezone(new \DateTimeZone('UTC'));
+        $endDate->setTimestamp($now);
+        $endDate->modify("midnight +{$daysInAdvance} days");
+
+        $qb = $this->_em->createQueryBuilder();
+        $exp = $qb->expr();
+
+        $qb->select('DISTINCT offering')->from(Offering::class, 'offering')
+            ->join('offering.session', 'session')
+            ->join('session.course', 'course')
+            ->join('course.school', 'school')
+            ->where($exp->andX(
+                $exp->gte('offering.startDate', ':startDate'),
+                $exp->lt('offering.startDate', ':endDate')
+            ))
+            ->andWhere($qb->expr()->in('school.id', ':schools'))
+            ->andWhere($qb->expr()->eq('session.published', true))
+            ->andWhere($qb->expr()->eq('course.published', true))
+            ->andWhere($qb->expr()->in('school.id', ':schools'))
```

Comment:
duplicates the `and where` clause three lines up, please remove.

Reply:
Nice catch! Fixed.